homeassistant/components/plugwise/climate.py

```diff
@@ -0,0 +1,310 @@
+        self._port = port
+        self._temperature = None
+        self._current_temperature = None
+        self._outdoor_temperature = None
+        self._state = None
+        self._active_schema = None
+        self._previous_schema = None
+        self._preset_mode = None
+        self._preset_modes = []
+        self._min_temp = min_temp
+        self._max_temp = max_temp
+        self._hvac_modes = ATTR_HVAC_MODES
+        self._hvac_mode = None
+
+        _LOGGER.debug("Initializing API")
+        import haanna
+        self._api = haanna.Haanna(self._username, self._password,
+                                  self._host, self._port)
+        try:
+            self._api.ping_anna_thermostat()
+        except Exception:
+            _LOGGER.error("Unable to ping, platform not ready")
+            raise PlatformNotReady
+        _LOGGER.debug("platform ready")
+        self.update()
+
+    @property
+    def should_poll(self):
+        """Return that we will require polling."""
+        return True
+
+    @property
+    def state(self):
+        """Return the current state."""
+        return self._state
+
+    @property
+    def name(self):
+        """Return the name of the thermostat, if any."""
+        return self._name
+
+    @property
+    def icon(self):
+        """Return the icon to use in the frontend."""
+        return DEFAULT_ICON
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return SUPPORT_FLAGS
+
+    @property
+    def device_state_attributes(self):
+        """Return the device specific state attributes."""
+        _attributes = {}
+        _attributes['outdoor_temperature'] = self._outdoor_temperature
+        _attributes['heating_action'] = self._heating_action
+        _attributes['available_schemas'] = self._available_schemas
+        _attributes['active_schema'] = self._active_schema
+        _attributes['previous_schema'] = self._previous_schema
+        return _attributes
+
+    def update(self):
+        """Update the data from the thermostat."""
+        _LOGGER.debug("Update called")
+        domain_objects = self._api.get_domain_objects()
+        self._current_temperature = self._api.get_temperature(domain_objects)
+        self._temperature = self._api.get_target_temperature(domain_objects)
+        self._preset_mode = self._api.get_current_preset(domain_objects)
+        self._outdoor_temperature = self._api.get_outdoor_temperature(
+            domain_objects)
+        self._heating_action = _heating_action(self)
+        self._available_schemas = _available_schemas(self)
+        if self._active_schema != _active_schema(self):
+            self._previous_schema = self._active_schema
+            self._active_schema = _active_schema(self)
+        self._hvac_mode = _hvac_mode(self)
+        self._preset_modes = []
+        for preset in [*self._api.get_presets(domain_objects)]:
+            self._preset_modes.append(preset)
```

Comment:
```
self._preset_modes = self._api.get_presets(domain_objects)??
```

Reply:
Adjusted it into the property definition with list(dict.keys())